DXMainClient/DXGUI/Multiplayer/CnCNet/PrivateMessagingWindow.cs

```diff
@@ -339,6 +335,17 @@ private void LbUserList_RightClick(object sender, EventArgs e)
             }
 
             playerContextMenu.Items[0].Text = cncnetUserData.IsFriend(lbUserList.SelectedItem.Text) ? "Remove Friend" : "Add Friend";
+            if ((bool)lbUserList.SelectedItem.Tag)
+            {
+                IRCUser iu = connectionManager.UserList.Find(u => u.Name == lbUserList.SelectedItem.Text);
+                playerContextMenu.Items[1].Text = cncnetUserData.IsIgnored(iu.Ident) ? "Unblock" : "Block";
+                playerContextMenu.Items[1].Selectable = true;
+            }
+            else
+            {
+                playerContextMenu.Items[1].Text = "Toggle Block";
+                playerContextMenu.Items[1].Selectable = false;
```

Comment:
Would be clearer to just hide this instead. Set `Visible = false` or preferably supply a function reference to the visibility checker for `AddItem` .